src/ui/liferea_shell.c

```diff
@@ -1352,15 +1352,28 @@ liferea_shell_present (void)
 	gtk_window_present (shell->window);
 }
 
+static gboolean
+liferea_shell_window_is_on_other_desktop(GdkWindow *gdkwindow)
+{
+#ifdef GDK_WINDOWING_X11
+	return GDK_IS_X11_DISPLAY (gdk_window_get_display (gdkwindow)) &&
+	    (gdk_x11_window_get_desktop (gdkwindow) !=
+	     gdk_x11_screen_get_current_desktop (gdk_window_get_screen (gdkwindow)));
+#else
+	return FALSE;
+#endif
+}
+
 void
 liferea_shell_toggle_visibility (void)
 {
 	GtkWidget *mainwindow = GTK_WIDGET (shell->window);
 	GdkWindow *gdkwindow = gtk_widget_get_window (mainwindow);
 
-	if (gdk_x11_window_get_desktop (gdkwindow) !=
-	    gdk_x11_screen_get_current_desktop (gdk_window_get_screen (gdkwindow))) {
+	if (liferea_shell_window_is_on_other_desktop (gdkwindow)) {
+#ifdef GDK_WINDOWING_X11
 		gdk_x11_window_move_to_current_desktop (gdkwindow);
```

Comment:
Right now, `liferea_shell_window_is_on_other_desktop` imply `GDK_IS_X11_DISPLAY` but to be future proof, I think it would be better to check here too.

Reply:
Sounds good -- I went back and forth on this one, but I think that's a good call.